### PR TITLE
Add `pytest` config and test `debiasing/preprocess_data.py`

### PR DESCRIPTION
**Pull request by griff-rees**
This branch was initially forked from `three-cities` to provide test infrastructure for `python` components within `debiasing`.

Tests can be run from `clim-recal/python` with the `conda` environment activated:

```console
$ cd cli-recal/python
$ conda activate clim-recal
$ pytest
....................                                                                                                                                                                
20 passed in 7.01s
```

Test configuration is currently managed within `clim-recal/python/.pytest.conf`. Two types of tests are currently included:

- Test files in `python/tests/` with file names prefixed by `test_` (currently only `clim-recal/python/tests/test_debiasing.py`)
- [`doctests`](https://docs.python.org/3/library/doctest.html) via `docstring` console sections. Examples are in `clim-recal/python/utils.py`.

### CODE REVIEW

**Review by gmingas — CHANGES_REQUESTED** [review]
This looks really good and also see comments below. 

Additionally, could we add instructions on how to run tests in the `README.md`? Just a simple few lines similar to the PR description.

File: python/tests/test_debiasing.py [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+        """Default method_1 option."""
+        return cls.QUANTILE_DELTA_MAPPING.value
+
+    @classmethod
+    def default_method_2(cls) -> str:
+        """Default method_2 option."""
+        return cls.VARIANCE_SCALING.value
+
+
+PROCESSESORS_DEFAULT: Final[int] = 32
```
[the remark this answers is not in the document]
Comment (griff-rees) [reply python/tests/test_debiasing.py:92]: Sure, it was a copy of the config for the server as best I could understand. Perhaps 2 for safety?

File: python/tests/test_debiasing.py [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+
+@pytest.mark.server
+@pytest.mark.slow
+@pytest.mark.parametrize(
+    "city",
+    (
+        None,
+        "Glasgow",
+    ),
+)
```
[the remark this answers is not in the document]
Comment (griff-rees) [reply python/tests/test_debiasing.py:697]: Cool. The current config tests two city cases:

- `Manchester` (the default in that config, hence `None`, and by extension that's a test of defaults
- `Glasgow`

Is there a particular set of parameter combinations you'd like to add?

File: python/tests/test_debiasing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmingas) [reply python/tests/test_debiasing.py:92]: Sounds good

File: python/tests/test_debiasing.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gmingas) [reply python/tests/test_debiasing.py:697]: I would probably add the equivalent of (London, 06, rainfall, delta_method) to test a few things at the same time, which should be enough for now

File: bash/ubuntu_install.sh
[the remark this answers is not in the document]
Comment (griff-rees) [reply bash/ubuntu_install.sh:1]: Apologies: having looked it over again, it's actually what I used to deploy the configuration on the `azure` box, in the hope that could be automated (especially if we can't use `docker`) so actually I think this should be included in this branch. Would you like to review this?

[the remark this answers is not in the document]
Comment (gmingas) [reply bash/ubuntu_install.sh:1]: Sure, happy to keep it in that case

File: python/tests/test_debiasing.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (griff-rees) [reply python/tests/test_debiasing.py:697]: Ok. I've got a draft commit. Logging will likely differ significantly for `rainfall` and `delta_method` so those portions of the test are non-trivial to alter. I'm putting in an example commit for review, but will need to connect to the server again to test.

[the remark this answers is not in the document]
Comment (griff-rees) [reply python/tests/test_debiasing.py]: Addressed in 52f7ac8c865d9b6d803e628dedd7ab3161740560

File: bash/ubuntu_install.sh
[the remark this answers is not in the document]
Comment (griff-rees) [reply bash/ubuntu_install.sh:1]: Cool thanks.

**Review by gmingas — APPROVED** [review]
Looks good!

**Pull request merged**